pkg/master/thirdparty_controller.go

```diff
@@ -37,8 +37,21 @@ func makeThirdPartyPath(group string) string {
 	return thirdpartyprefix + "/" + group
 }
 
-func getThirdPartyGroupName(path string) string {
-	return strings.TrimPrefix(strings.TrimPrefix(path, thirdpartyprefix), "/")
+func makeThirdPartyVersionedPath(group, version string) string {
+	if len(group) == 0 {
+		return thirdpartyprefix
+	}
+	return fmt.Sprintf("%s/%s/%s", thirdpartyprefix, group, version)
+}
+
+func getThirdPartyGroupKind(path string) (string, string) {
+	parts := strings.Split(path, "/")
+	if len(parts) == 4 {
+		return parts[2], parts[3]
+	} else if len(parts) == 5 {
```

Comment:
length 5 would be the Resource URL with the version name, which is also registered as a webservice.

Example of length 5 : /apis/example.com/v1/foos
Example of length 4: /apis/example.com/v1/foos/